### ISSUE TITLE
Possible Conflict Causing Crash Between the Translocator and the Fluid Tanks from Create

### ISSUE THREAD

**Issue by SwissxCore** [TeamPneumatic/pnc-repressurized#1199]
### Describe the bug

While in a server, I attached a translocator to a fluid tank from the Create mod, and upon attempting to remove it using the pneumatic wrench, the server crashed and everyone was returned to the menu.

I was then able to recreate the bug in a single player world, in which the crash actually closed the entire client instead of returning me to the menu.

### How to reproduce the bug

1. Place a fluid tank from the Create mod, the size doesn't matter, nor does the contents.
2. Attach a translocator from PneumaticCraft to the side of the fluid tank.
3. Attempt to disconnect the translocator using the pneumatic wrench by shift+right clicking.

### Expected behavior

Upon attempting to disconnect the translocator, it is expected to simply pop off as an item in the world, ready to be picked up by the player.

### Additional details

For your convenience, I have tried to recreate the bug in 1.20.1 of Minecraft, using Forge version 47.1.44, but I was unable to do so; instead, the translocator doesn't seem to have a hitbox at all and cannot be interacted with using the pneumatic wrench; the pneumatic wrench will instead disassemble the fluid tank, ignoring the translocator, and the translocator will simply drop as an item since it is no longer attached to a surface.

This is a bug for an earlier version of PneumaticCraft, but I noticed you still allow 1.18 bugs if they're critical, and I think this fits the bill.

I unfortunately haven't checked to see if the bug occurs with the translocator on any other surface, only with it on the fluid tanks by create, but the pneumatic wrench appears to break the create fluid tanks without crashing if there is no translocator, it's only when trying to remove the translocator that it crashes.

And lastly, I am able to break the translocator by viciously beating it until it drops as an item without the game crashing, implying it has something to do with the interaction between it and the pneumatic wrench.

(Apologies, I can't access Pastebin, or any website with Cloudflare, and I can't figure out why... It's been months now; so, here's a Google Docs link to the crash report)

### Which Minecraft version are you using?

1.18 - critical bugs only

### Which version of PneumaticCraft: Repressurized are you using?

3.6.0-19

### Crash log

https://docs.google.com/document/d/1hm4DfMdCTdaiEgLDM-1759PUkL9mcD5ufcHr05kkJUM/edit?usp=sharing

**Comment by SwissxCore** [comment]
You know what, I just saw an earlier issue that this probably pertains to; when I did the search, I searched specifically for "translocator", and so I missed this; but I was just now looking down the list at some other reports and saw it.
https://github.com/TeamPneumatic/pnc-repressurized/issues/1187
I suppose until Tinker's Construct moves out of 1.18, I'll just have to avoid using Create and PneumaticCraft too closely together.

**Comment by desht** [comment]
What's a "translocator" in the context of PNC?  The mod adds no such block or item.

**Comment by SwissxCore** [comment]
> What's a "translocator" in the context of PNC? The mod adds no such block or item.

Apologies, I'm having a hard time loading a world now, for some reason PneumaticCraft is causing me to crash when joining a world. I threw the crash log into ChatGPT and it spat this back out at me:

```
java.lang.NullPointerException: Cannot invoke "com.google.gson.JsonElement.getAsJsonArray()" because the return value of "com.google.gson.JsonObject.get(String)" is null
	at me.desht.pneumaticcraft.common.config.subconfig.ArmorFeatureStatus.readFromJson(ArmorFeatureStatus.java:61) ~[pneumaticcraft-repressurized-1.18.2-3.6.0-19.jar%2365!/:1.18.2-3.6.0-19] {re:classloading}
```

I simply removed the mod to test this bug in 1.20.1, which was null, and now I've dragged the 1.18.2 version back into the mods folder and I'm getting this error... Nothing should have changed. I just don't know what to do anymore, I'm out of ideas.

Maybe I just got the name wrong, so I'll try to describe it:
It's a small block that you can place on the side of a container, be it a chest or tank, and it will slowly transfer the contents at 1 item (or 1 bucket) per 2 seconds into an adjacent container.

**Comment by desht** [comment]
Oh, the Transfer Gadget.

Anyway, the actual bug in 1.18.2 is fixed in dev now and I'll get a release out soon (yes I'm still fixing crashes for 1.18 if they can negatively impact a whole server).

The config problem you have above sounds like a corrupted `config/pneumaticcraft/ArmorFeatureStatus.cfg` file - try removing that and restarting the instance - it will generate a new default file.  Update: oh - if you loaded PNC 1.20, it will have changed the format of that file.  So yes definitely delete and have 1.18 recreate it.

**Comment by SwissxCore** [comment]
Desht, you are a champion, deleting and allowing that config file to regenerate worked like a charm. I sincerely appreciate the effort you're putting into this, even with older versions of the game, and I apologise for the confusion I caused with the transfer gadget, even know I have no idea why I called it a translocator.

I'll close with this since the bug has been reported and that config issue I was having has been fixed. Thanks again.
